### PR TITLE
move try block out of AST_Try, so block scopes work correctly

### LINKED ISSUE #1107: Variable in `finally` branch capture the wrong one
**Bug report or Feature request?**
Bug report

**Version (complete output of `terser -V` or specific git commit)**
4.3.1

**Complete CLI command or `minify()` options used**
```js
{
  module: true,
  compress: {},
  mangle: {},
  output: {},
  parse: {},
  rename: {},
}
```

**`terser` input**
```js
function foo() {
  let bar = 23;
  
  try {
    const bar = 12;
    console.log(bar);
  } finally {
  	console.log(bar);
  }
}

foo();
```

**`terser` output or error**
```js
!function(){try{const o=12;console.log(o)}finally{console.log(o)}}();
```

**Expected result**
```js
!function(){try{const o=12;console.log(o)}finally{console.log(23)}}();
```

### PR DESCRIPTION
**Pull request by fabiosantoscode**
Fixes #1107

Opening a PR because I'd appreciate more eyeballs on this.

### CODE REVIEW

File: lib/output.js [hunk 1]
```diff
@@ -1555,7 +1555,7 @@ function OutputStream(options) {
     DEFPRINT(AST_Try, function(self, output) {
         output.print("try");
         output.space();
-        print_braced(self, output);
+        print_braced(self.body, output);
```
Comment (jridgewell) [inline lib/output.js:1558]: Ditto.

**Review by jridgewell — CHANGES_REQUESTED** [review]
Need to add the test case from https://github.com/terser/terser/issues/1107.

File: lib/ast.js [hunk 2]
```diff
@@ -120,6 +120,13 @@ class AST_Token {
         Object.seal(this);
     }
 
+    [Symbol.for("nodejs.util.inspect.custom")](_depth, options) {
```
[the remark this answers is not in the document]
Comment (fabiosantoscode) [reply lib/ast.js:123]: This is for console.log - ing nodes. Before, we used to get `start` and `end` properties that occupied a lot of lines. This lets us show just `[AST_Token 'return' at 123:10]` which only takes one line and has most information anyway.

[the remark this answers is not in the document]
Comment (fabiosantoscode) [reply lib/ast.js:123]: image

It looks like this.

File: lib/compress/inference.js [hunk 3]
```diff
@@ -298,7 +298,7 @@ export function is_nullish(node, compressor) {
             || any(this.body, compressor);
     });
     def_has_side_effects(AST_Try, function(compressor) {
-        return any(this.body, compressor)
+        return any(this.body.body, compressor)
```
[the remark this answers is not in the document]
Comment (fabiosantoscode) [reply lib/compress/inference.js:301]: Yup let me do that

File: lib/output.js (hunk 1 above)
Comment (fabiosantoscode) [reply lib/output.js:1558]: I had to make AST_Block printable but that will be useful for later.

File: lib/ast.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jridgewell) [reply lib/ast.js:123]: We might be able to define a generic one on the base class?

**Comment by jridgewell** [comment]
I tested #1107, and while it passes (we should add the test case), it doesn't produce an optimal output:

```js
!function(){try{const bar=12;console.log(bar)}finally{console.log(23)}}();
```

That should have inlined `bar` into `console.log(12)`, right?

File: lib/ast.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabiosantoscode) [reply lib/ast.js:123]: I think we can have both. But since most of the output when you console.log a node, is AST_Token, that's what I tackled first.

**Comment by fabiosantoscode** [comment]
@jridgewell 

> That should have inlined bar into console.log(12), right?

It's a const in a try block so that's two things Terser doesn't like to mess with :)

**Comment by fabiosantoscode** [comment]
I re-did this by hand in https://github.com/terser/terser/commit/627a4337a22df528442f693b4b3216abb2b6afae instead of merging the conflicts.

**Pull request closed without merging**